docs/GDPs/gdp-0002-regions.rst

```diff
@@ -0,0 +1,275 @@
+2. Add parsing tests
+3. Add code generation support
+4. Code generation tests
+5. Create at least one demo that incorporates this feature
+
+
+FV3 Example
+-----------
+
+.. code-block:: Fortran
+
+    subroutine divergence_corner(u, v, ua, va, divg_d, ...)
+
+    ! arguments
+    real :: ua(isd:ied, jsd:jed)          ! cell-center
+    real :: va(isd:ied, jsd:jed)          ! cell-center
+    real :: u(isd:ied, jsd:jed+1)         ! staggered in y-direction
+    real :: v(isd:ied+1, jsd:jed)         ! staggered in x-direction
+    real :: divg_d(isd:ied+1, jsd:jed+1)  ! corner (staggered both in x- and y-direction)
+
+    ! locals
+    real :: uf(is-2:ie+2, js-1:je+2)      ! staggered in y-direction
+    real :: vf(is-1:ie+2, js-2:je+2)      ! staggered in y-direction
+
+    ! indices
+    integer :: is,  ie,  js,  je   ! compute domain
+    integer :: isd, ied, jsd, jed  ! data domain = compute domain + halo zone
+
+    is2 = max(2, is)         ! restrict computation to exclude west-edge
+    ie1 = min(npx-1, ie+1)   ! restrict computation to exclude east-edge
+
+    do j = js, je+1
+      if (j == 1 .or. j == npy) then
+        do i = is-1, ie+1
+          uf(i,j) =
+            u(i,j)*dyc(i,j)*0.5*(sin_sg(i,j-1,4) + sin_sg(i,j,2))
+        end do
+      else
+        do i = is-1, ie+1
+          uf(i,j) = &
+              (u(i,j) - 0.25*(va(i,j-1) + va(i,j))*(cos_sg(i,j-1,4) + cos_sg(i,j,2)))  &
+                                      *dyc(i,j)*0.5*(sin_sg(i,j-1,4) + sin_sg(i,j,2))
+        end do
+      end if
+    end do
+
+    do j = js-1, je+1
+      do i = is2, ie1     ! inner domain (full compute domain for ranks without edges)
+        vf(i, j) = &
+          (v(i,j) - 0.25*(ua(i-1,j) + ua(i, j))*(cos_sg(i-1,j,3) + cos_sg(i,j,1)))  &
+                                *dxc(i,j)*0.5*(sin_sg(i-1,j,3) + sin_sg(i,j,1))
+      end do
+      if (is == 1) &      ! west-edge
+        vf(1, j) = &
+          v(1, j)*dxc(1, j)*0.5*(sin_sg(0, j, 3) + sin_sg(1, j, 1))
+      if (ie+1 == npx) &  ! east-edge
+        vf(npx, j) = &
+          v(npx, j)*dxc(npx,j)*0.5*(sin_sg(npx-1, j, 3) + sin_sg(npx, j, 1))
+    end do
+
+    do j=js,je+1
+      do i=is,ie+1
+        divg_d(i,j) = vf(i,j-1) - vf(i,j) + uf(i-1,j) - uf(i,j)
+      end do
+    end do
+
+    if (gridstruct%sw_corner) &
+      divg_d(1,    1) = divg_d(1,    1) - vf(1,    0)
+    if (gridstruct%se_corner) &
+      divg_d(npx,  1) = divg_d(npx,  1) - vf(npx,  0)
+    if (gridstruct%ne_corner) &
+      divg_d(npx,npy) = divg_d(npx,npy) + vf(npx,npy)
+    if (gridstruct%nw_corner) &
+      divg_d(1,  npy) = divg_d(1,  npy) + vf(1,  npy)
+
+    do j=js,je+1
+      do i=is,ie+1
+        divg_d(i,j) = rarea_c(i,j) * divg_d(i,j)
+      end do
+    end do
+
+.. code-block:: python
+
+    @gtscript.stencil(...)
+    def divergence_corner(...):
+        from __externals__ import istart, iend, jstart, jend
```

Comment:
Can you explain the precise meaning and/or expected values for these splitters (istart, iend, ...)?

Reply:
I added the code from above. It's the same thing. Thanks for catching that!